Submitted By: Alexander E. Patrakov (semzx at newmail dot ru)
Origin: Alexander E. Patrakov
Date: 2003-11-22
Initial Package Version: 2.85
Description: Without this patch, init sometimes clobbers environment variables
set by kernel (such as TERM) when trying to change process title for ps
due to out-of-bounds access to the command line arguments.

diff -ur sysvinit-2.85/src/init.c sysvinit-2.85-fixed/src/init.c
--- sysvinit-2.85/src/init.c	2003-04-15 17:16:41.000000000 +0600
+++ sysvinit-2.85-fixed/src/init.c	2003-11-22 19:46:03.000000000 +0500
@@ -432,7 +432,7 @@
 	len = vsnprintf(buf, sizeof(buf), fmt, ap);
 	va_end(ap);
 
-	memset(argv0, 0, maxproclen + 1);
+	memset(argv0, 0, maxproclen);
 	strncpy(argv0, buf, maxproclen - 1);
 
 	return len;
